packages/salesforcedx-vscode-core/src/commands/forceSourceDeploySourcePath.ts

```diff
@@ -51,21 +56,42 @@ export class LibraryDeploySourcePathExecutor extends DeployExecutor<
   }
 }
 
-export class LibraryPathsGatherer implements ParametersGatherer<string[]> {
-  private uris: vscode.Uri[];
-  public constructor(uris: vscode.Uri[]) {
-    this.uris = uris;
-  }
-  public async gather(): Promise<ContinueResponse<string[]>> {
-    const sourcePaths = this.uris.map(uri => uri.fsPath);
-    return {
-      type: 'CONTINUE',
-      data: sourcePaths
-    };
+export const forceSourceDeploySourcePath = async (
+  sourceUri: vscode.Uri,
+  uris: vscode.Uri[]
+) => {
+  // In the case that only one file is deployed,
+  // uris contains a single path - the same as sourceUri
+  // and isn't undefined.
+  if (uris && uris.length > 1) {
+    await forceSourceDeployMultipleSourcePaths(uris);
+  } else {
+    await forceSourceDeploySingleSourcePath(sourceUri);
   }
-}
+};
+
+export const forceSourceDeployMultipleSourcePaths = async (uris: vscode.Uri[]) => {
+  const messages: ConflictDetectionMessages = {
+    warningMessageKey: 'conflict_detect_conflicts_during_deploy',
+    commandHint: input => {
+      return new SfdxCommandBuilder()
+        .withArg('force:source:deploy')
+        .withFlag('--sourcepath', input)
+        .build()
+        .toString();
+    }
+  };
+
+  const commandlet = new SfdxCommandlet<string | string[]>(
+    new SfdxWorkspaceChecker(),
+    new LibraryPathsGatherer(uris),
+    new LibraryDeploySourcePathExecutor(),
+    new TimestampConflictChecker(false, messages)
+  );
+  await commandlet.run();
+};
 
-export async function forceSourceDeploySourcePath(sourceUri: vscode.Uri) {
+export const forceSourceDeploySingleSourcePath = async (sourceUri: vscode.Uri) => {
```

Comment:
Randi & I discovered that in order to get the stubs and spies to work, the syntax of the function declarations of the static functions needed to be changed.